src/csv_reader/csv_reader.c

```
@@ -72,8 +100,21 @@ static void _map_cell_cb
-	// append cell to current row
+	if(unlikely(reader->col_idx >= reader->col_count)) {
+		ErrorCtx_SetError("CSV row of unexpected length");
+		return;
+	}
+
 	SIValue key = reader->columns[reader->col_idx++];
+
+	// empty cell is treated as an empty string
+	if(unlikely(n == 0)) {
+		ASSERT(data == NULL);
+		// do not add key to map for missing cells
```

cppcheck (2.10-2)

[error] Common realloc mistake

(memleakOnRealloc)